pyingest/parsers/jats.py

```diff
@@ -0,0 +1,231 @@
+#!/usr/bin/env python
+
+import bs4
+from default import BaseBeautifulSoupParser
+from pyingest.config.config import *
+
+
+class NoSchemaException(Exception):
+    pass
+
+class WrongSchemaException(Exception):
+    pass
+
+class UnparseableException(Exception):
+    pass
+
+class JATSParser(BaseBeautifulSoupParser):
+
+    def __init__(self):
+        pass
+
+    def _detag(self, r, tags_keep, **kwargs):
+        newr = bs4.BeautifulSoup(unicode(r),'lxml')
+        tag_list = list(set([x.name for x in newr.find_all()]))
+        for t in tag_list:
+            if t in JATS_TAGS_DANGER:
+                newr.find(t).decompose()
+            elif t in tags_keep:
+                newr.find(t).contents
```

Comment:
I don't think this does anything to the parsed tree?

Reply:
I verified that it pulls out all "DANGER" tags in their entirety, and returns the verbatim contents of the tag object otherwise.

Reply:
what I meant was: what is the purpose of `newr.find(t).contents`?  It didn't look to me like calling this method was doing anything to modify the `newr` object, in which case why not just have `pass`?